lldb/tools/debugserver/source/MacOSX/MachProcess.mm

```diff
@@ -1573,6 +1576,8 @@ static uint64_t bits(uint64_t value, uint32_t msbit, uint32_t lsbit) {
 }
 
 bool MachProcess::Interrupt() {
+  PTHREAD_MUTEX_LOCKER(locker, m_exception_and_signal_mutex);
+
```

Comment:
Would it be better to put this mutex lock after the IsRunning(state) on line 1582.  You aren't doing anything that requires the mutex till then, and in general it's better to scope this as narrowly as possible.

Reply:
I was worried about lock order inversion between the exception mutex and the state mutex. That said, how we're protecting GetState is already questionable (i.e. nothing guarantees the state hasn't changed between the function returned you read the value) so I can tighten the scope and we can fix the issue if it comes up when we try to fix the state locking.